src/corelibs/U2View/src/ov_msa/PhyTrees/MSAEditorTreeViewer.cpp

```diff
@@ -50,6 +50,12 @@ MSAEditorTreeViewer::MSAEditorTreeViewer(const QString &viewName, GObject *obj,
       msaTreeViewerUi(nullptr) {
 }
 
+MSAEditorTreeViewer::~MSAEditorTreeViewer() {
+    if (editor != nullptr && isSyncModeEnabled()) {
```

Comment:
Don't like it. Destructors are supposed to clear the captured memory, not do some post-actions. I think, it's better to make the slot `objectIsAbout2BeDestroyed`, connect it to [QObject::destroyed](https://doc.qt.io/qt-5/qobject.html#destroyed) and mode this part of code to this slot.

Reply:
Ok, will try to replace it with destroyed() signal.

Reply:
I prepared a solution with destroy() signal handling, checked that it worked, but later decided to roll it back (see the history of commits) 

The way to solve the problem with destroy() adds a new state management logic (subscribe/unsubscribe/do-not subscribe twice...) and adds a lot of boilerplate code.

For example the original solution with the destructor requires 3 lines against 10+ in the new code and technically works absolutely the same.

I think the final cleanup is out of scope today because the code will be reworked in the next release: moved away from Sequence Area to MSA Editor, etc. So we will have a second change to discuss it.

Reply:
I disagree and think, that the solution with `destroy()` was good enough. I won't argue about your solution cuz don't wan't to.

Reply:
I agree with your point that 'destructors are evil'. 

But the problem is that the alternative with connect/disconnect is not better at all: it is still a manual management of life-range callbacks using very verbose QT signal/slots. 

I accept the problem you mentioned and will try to address it in a better way during the first spring of the next release. For now I'm keeping the smaller/simpler solution.